lamindb/dev/db/_add.py

```diff
@@ -81,28 +81,84 @@ def add(  # type: ignore
             return results
         else:
             records = [model(**fields)]
-    for record in records:
-        if isinstance(record, DObject) and hasattr(record, "_local_filepath"):
-            upload_data_object(record, use_fsspec=use_fsspec)
+
     if session is None:  # assume global session
         session = settings.instance.session()
         settings.instance._cloud_sqlite_locker.lock()
         close = True
     else:
         close = False
+
+    added_records = []
+    raise_error = False
     for record in records:
-        session.add(record)
-    session.commit()
-    for record in records:
+        # commit metadata to database
+        prepare_filekey_metadata(record)
+        try:
+            session.add(record)
+            session.commit()
```

Comment:
This is the only thing that should be contained in the try except, I think!

Reply:
Agreed and implemented. Errors with `session.add()` will be raised unhandled with the full traceback. Nothing will have been commited to the DB or uploaded to storage at that point.